### ISSUE TITLE
Duplicate initialization on HMR with vite

### ISSUE THREAD

**Issue by zynth17** [andrewcourtice/harlem#24]
**Describe the bug**
i'm using vue with typescript and vite. the problem is everytime i'm updating a file where i call a mutation in that component
the HMR always failed to reload because harlem keep trying to create a new store

**My Component**
```vue
<script lang="ts">
import {
computed,
ComputedRef,
defineComponent,
inject,
onMounted,
PropType,
} from 'vue';
import { getError, getLoadingButton } from '@/store';
import { useFile } from '@/components/hooks/useFile';
import { decreaseTimer, getTimer, getUser, setTimer } from '@/store/user';

export default defineComponent({
name: 'ProfileForm',
props: {
	user: {
		type: Object as PropType<User>,
		required: true,
	},
},
setup: function () {
	const { selectedFile, urlPreview, onChangeFile } = useFile();
	const editProfile = inject<(file: File) => void>('editProfile') as (
		file: File,
	) => void;
	const resendEmail = inject<() => void>(
		'resendEmailVerification',
	) as () => void;
	const openDialog = inject<() => void>('openDialog') as () => void;

	const avatarUrl = computed<string | undefined>(() => {
		const user = getUser.value;
		if (!user?.avatar_thumbnail && !urlPreview.value) {
			return '/assets/images/input-avatar.png';
		} else if (urlPreview.value) {
			return urlPreview.value;
		} else {
			return user?.avatar_thumbnail;
		}
	});
	let timer: number;
	const timerFn = () => {
		timer = setInterval(() => {
			if (getTimer.value.time <= 0) {
				clearInterval(timer);
				setTimer({ isCountdown: false, time: 60 });
			} else {
				decreaseTimer();
				setTimer({
					isCountdown: true,
					time: getTimer.value.time,
				});
			}
		}, 1000);
	};
	const resendEmailVerification = () => {
		timerFn();
		resendEmail();
	};
	onMounted(() => {
		if (getTimer.value.isCountdown) timerFn();
	});
	return {
		onChangeFile,
		selectedFile,
		avatarUrl,
		editProfile,
		openDialog,
		resendEmailVerification,
		timer: getTimer as ComputedRef<{
			isCountdown: boolean;
			time: number;
		}>,
		getUser,
		loading: getLoadingButton,
		errors: getError,
	};
},
});
</script>
```

**My User Store**
```typescript
import { createStore } from '@harlem/core';
import { ComputedRef } from 'vue';
import { RouteStatus } from '@/routes';

interface State {
	user: User | null;
	status: RouteStatus.USER | RouteStatus.SELLER;
	timer: {
		isCountdown: boolean;
		time: number;
	};
}

const STATE = <State>{
	user: null,
	status: RouteStatus.USER,
	timer: {
		isCountdown: false,
		time: 60,
	},
};

const { getter, mutation } = createStore('user', STATE);

export const setUser = mutation<User>('setUser', (state, payload) => {
	state.user = payload;
	localStorage.setItem('user', JSON.stringify(payload));
});

export const removeUser = mutation('removeUser', state => {
	localStorage.removeItem('user');
	state.user = null;
});

export const getUser = getter('getUser', state => {
	const localUser: string | null = localStorage.getItem('user');
	if (localUser) {
		setUser(JSON.parse(localUser) as User);
	}
	return state.user;
});

export const isAuthorized = getter('isAuthorized', () =>
	checkAuthorization(getUser),
);

export const checkAuthorization = (
	user: ComputedRef<User | null>,
): user is ComputedRef<User> => {
	return user.value !== null;
};

export const setSeller = mutation<boolean>('setSeller', (state, payload) => {
	if (payload) state.status = RouteStatus.SELLER;
	else state.status = RouteStatus.USER;
});

export const isSeller = getter('isSeller', state => {
	return state.status === RouteStatus.SELLER;
});
export const getStatus = getter('getStatus', state => {
	return state.status;
});

export const getToken = getter('getToken', state => {
	return state.user?.token;
});

export const setTimer = mutation<{ isCountdown: boolean; time: number }>(
	'setTimer',
	(state, payload) => {
		localStorage.setItem('timer', JSON.stringify(payload));
		state.timer = payload;
	},
);

export const decreaseTimer = mutation('decreaseTimer', state => {
	state.timer.time--;
});
export const getTimer = getter('getTimer', state => {
	let localTimer: { isCountdown: boolean; time: number } | null = JSON.parse(
		<string>localStorage.getItem('timer'),
	);
	if (localTimer) {
		setTimer(localTimer);
	}
	return state.timer;
});
```

**To Reproduce**

Steps to reproduce the behaviour:
1. install vue with vite
2. create a store
3. start the server
4. call a mutation from component
5. do some change on the component
6. See error when the hmr try to reloading the page

**Expected behaviour**
Supposedly the store need to reset the instance and create a new one when hmr reload

**Screenshots**
![image](https://user-images.githubusercontent.com/45350572/112290000-653ceb00-8cc1-11eb-942a-5482e2c2e558.png)

**Version (please complete the following information):**
 - Harlem: [e.g. 1.1.1]
 - Vue: [e.g. 3.0.5]
 - Runtime Environment: Version 89.0.4389.90 (Official Build) (Windows 10 64-bit)

**Comment by andrewcourtice** [comment]
Hi @zynth17 

Thanks for sending this one through. I am unable to reproduce the error using the steps you provided, however I can reproduce the problem when I make a change to a mutation file (.ts) and save it as opposed to a component file (.vue).

Can you please confirm if the issue is specifically happening when you change `vue` files or when you change `ts` files that contain imports to the store?

**Comment by zynth17** [comment]
hi thankyou for responding! yes this issue is happening only when i change (.vue) file that contain import to the store.

**Comment by andrewcourtice** [comment]
@zynth17 This is fixed in v1.2.3.

Harlem will now log a warning in the console when HMR replaces a store/mutation/getter as opposed to throwing an error (only in dev environments).

**Comment by zynth17** [comment]
Thankyou so much for the quick fix!
